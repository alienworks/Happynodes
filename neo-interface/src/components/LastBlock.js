import React, { Component } from 'react'
import { connect } from 'react-refetch'
import config from './config'

class LastTime extends Component {
    constructor(props) {
        super(props);
        this.state = {
            lastblocktime: 0,
            seconds: 0
        };
    }
    tick() {
        if (this.props.lastblock.fulfilled) {

            this.setState(() => ({
                lastblocktime: this.props.lastblock.value.lastblock,
                seconds: (this.state.lastblocktime > 0) ? ((new Date() - new Date(Number(this.state.lastblocktime) * 1000)) / 1000).toFixed(0) : 0
            }));
        }
    }

    componentDidMount() {
        this.interval = setInterval(() => this.tick(), 500);
    }

    componentWillUnmount() {
        clearInterval(this.interval);
    }
    render() {
        const { lastblock } = this.props

        if (lastblock.pending) {
            return (
                <br />
            );
        }


        return (
            <div className="top-block pink">

                <h3>{this.state.seconds}s ago</h3>


                <p>LAST BLOCK SEEN</p></div>
        );
    }
}

export default connect((props) => ({
    lastblock: {
        url: config.api_url.concat(`/lastblock`),
        refreshInterval: 1000
    }
}))(LastTime)