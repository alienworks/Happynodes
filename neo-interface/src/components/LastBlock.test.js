import React from 'react';
import ReactDOM from 'react-dom';
import LastBlock from './LastBlock';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LastBlock />, div);
    ReactDOM.unmountComponentAtNode(div);
});

describe("test against valid snapshot", ()=> {
    it("redner as expected", ()=> {
        const tree = renderer.create(<LastBlock lastblock={{value:{lastblock:1534363759223}}} />).toJSON();
        expect(tree).toMatchSnapshot();

    });
});

describe("test against valid snapshot", ()=> {
    it("redner as expected", ()=> {
        const tree = renderer.create(<LastBlock lastblock={{fulfilled:true, value:{lastblock:1534363759223}}} />).toJSON();
        expect(tree).toMatchSnapshot();

    });
});
  