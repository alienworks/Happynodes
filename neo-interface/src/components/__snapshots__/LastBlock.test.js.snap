// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test against valid snapshot redner as expected 1`] = `
<div
  className="top-block pink"
>
  <h2>
    0
    s ago
  </h2>
  <p>
    LAST BLOCK SEEN
  </p>
</div>
`;

exports[`test against valid snapshot redner as expected 2`] = `
<div
  className="top-block pink"
>
  <h2>
    0
    s ago
  </h2>
  <p>
    LAST BLOCK SEEN
  </p>
</div>
`;
